kotlinx-coroutines-core/build.gradle

```diff
@@ -205,12 +219,26 @@ task jvmStressTest(type: Test, dependsOn: compileTestKotlinJvm) {
     systemProperty 'kotlinx.coroutines.semaphore.maxSpinCycles', '10'
 }
 
+task jvmLincheckTest(type: Test, dependsOn: compileTestKotlinJvm) {
+    classpath = files { jvmTest.classpath }
+    testClassesDirs = files { jvmTest.testClassesDirs }
+    minHeapSize = '1g'
+    maxHeapSize = '6g' // we may need more space for building an interleaving tree in the model checking mode
+    include '**/*LincheckTest.*'
+    enableAssertions = true
+    testLogging.showStandardStreams = true
+    jvmArgs = ['--add-opens', 'java.base/jdk.internal.misc=ALL-UNNAMED',   // required for transformation
+               '--add-exports', 'java.base/jdk.internal.util=ALL-UNNAMED'] // in the model checking mode
+    systemProperty 'kotlinx.coroutines.semaphore.segmentSize', '2'
+    systemProperty 'kotlinx.coroutines.semaphore.maxSpinCycles', '1' // better for the model checking mode
+}
+
 task jdk16Test(type: Test, dependsOn: [compileTestKotlinJvm, checkJdk16]) {
     classpath = files { jvmTest.classpath }
     testClassesDirs = files { jvmTest.testClassesDirs }
     executable = "$System.env.JDK_16/bin/java"
     exclude '**/*LFStressTest.*' // lock-freedom tests use LockFreedomTestEnvironment which needs JDK8
-    exclude '**/*LCStressTest.*' // lin-check tests use LinChecker which needs JDK8
+    exclude '**/*LincheckTest.*' // Lincheck tests use LinChecker which needs JDK8
```

Comment:
That would also need renaming all the `xxxLCTest.kt` files to `xxxLinckeckTest.kt`. Actually, I suggest to revert the name change and leave all the classes named as `xxxLCTest`, using `LC` as an abbreviation for `Linckeck`. Otherwise, it is somewhat inconvenient, since the first time you open the source code you get `"Typo: In word 'Lincheck'` warning.

Reply:
I would prefer not to revert this change since the names were `xxxLCStressTest.kt` and we have stress testing and model checking modes in Lincheck -- this overlap in the "stress" word usages confuses a bit.